.github/workflows/pre-merge.yml

```
@@ -0,0 +1,55 @@
+name: Pre-merge
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --no-progress --non-interactive
```

```suggestion
  setup:
    runs-on: ubuntu-latest
    steps:
      - name: Set up Node.js
        uses: actions/setup-node@v4

      - name: Checkout repository
        uses: actions/checkout@v4

      - name: Install dependencies
        run: yarn install --frozen-lockfile --no-progress --non-interactive
```

After inlining, we can remove that job.